File: 02_ComparisonSorts/SelectionSort/SelectionSort.py
import random
 
"""
 creating an array of size n
 initializing the arry with random values
"""
def initializeArrayWithRandomNumbers(n):
    myList = []
    for i in range(n):
        myList.append(random.randint(0,n))
    return myList
	
def selectionSort(myList):
	for i in range(len(myList)-1):
		smallest = myList[i]
		smallpos = i
		for j in range(i+1, len(myList)):
			if myList[j] < smallest:
				smallest = myList[j]
				smallpos = j
		temp = myList[i]
		myList[i] = myList[smallpos]
		myList[smallpos] = temp
	return myList

size = 10
print("Unsorted Array")
myList = initializeArrayWithRandomNumbers(size)
print(myList)

print("**********Selection Sort**********")
myList = selectionSort(myList)
print(myList)

File: 02_ComparisonSorts/SelectionSort/SelectionSort.java
import java.io.*;
import java.util.*;

class SelectionSort {
    /*
     * creating an array of size n
     * initializing the arry with random values
     */
    public int[] initializeArrayWithRandomNumbers(int n) {
        int arr[] = new int[n];
        int i;
        Random rand = new Random();
        for (i = 0; i < n; ++i) {
            arr[i] = rand.nextInt(n);
        }
        return arr;
    }

    // Swaps the values at position1 and position2 in the array parameter and
    // returns the edited array
    public int[] swap(int[] arr, int position1, int position2) {
        int temporary = arr[position1];
        arr[position1] = arr[position2];
        arr[position2] = temporary;
        return arr;
    }// swap()

    public void printArray(int arr[]) {
        int i;
        for (i = 0; i < arr.length; ++i) {
            System.out.print(arr[i] + "\t");
        }
        System.out.println();
    }

    protected int[] selectionSort(int[] arr) {
        int smallest, i, j, SmallestPosition;
        // One by one move boundary of unsorted subarray
        for (i = 0; i < ((arr.length) - 1); ++i) {
            smallest = arr[i];
            SmallestPosition = i;
            // Find the minimum element in unsorted array
            for (j = i + 1; j < arr.length; ++j) {
                if (arr[j] < smallest) {
                    smallest = arr[j];
                    SmallestPosition = j;
                } // if block
            } // for j loop (inner)
              // Swap the found minimum element with the first element
            arr = swap(arr, i, SmallestPosition);
        } // for i loop (outer)
        return arr;
    }// selectionSort()
}

class MainSelection {
    public static void main(String[] args) {
        SelectionSort ss = new SelectionSort();
        int size = 10;
        int arrAns[];
        int arr[] = ss.initializeArrayWithRandomNumbers(size);
        System.out.println("Unsorted Array");
        ss.printArray(arr);

        // Bubble Sort
        System.out.println("**********Selection Sort**********");
        arrAns = ss.selectionSort(arr);
        ss.printArray(arrAns);
    }
}
